### PR TITLE
restart service on device restart,

### PR DESCRIPTION
**Pull request by spuday90**
Added Broadcast receiver to restart service on device boot

### CODE REVIEW

**Comment by spuday90** [comment]
@adeekshith check if autostart permission is redirecting to exact settings on your device

**Comment by adeekshith** [comment]
> @adeekshith check if autostart permission is redirecting to exact settings on your device

It is not requesting for any permissions. Should I perform any action for it to show up?

**Comment by adeekshith** [comment]
In this PR, it is not showing Watomatic in the settings while requesting "Notification access"

| This PR        | Main           |
| ------------- |:-------------:|
| Screenshot_1621919066  | Screenshot_1621919123 |

**Review by adeekshith — CHANGES_REQUESTED** [review]
Notification access permission not working on Android 11 (API 30)

**Comment by spuday90** [comment]
> Notification access permission not working on Android 11 (API 30)

for backward compatibility , we should keep enable service method for next few versions.

**Comment by spuday90** [comment]
> > @adeekshith check if autostart permission is redirecting to exact settings on your device
> 
> It is not requesting for any permissions. Should I perform any action for it to show up?

Changed it to show popup for existing users on resume of app

**Comment by adeekshith** [comment]
Did not test behavior for existing users yet but worked well on emulator for new install.
Restarting by itself for new users.

**Comment by adeekshith** [comment]
> > > @adeekshith check if autostart permission is redirecting to exact settings on your device
> > 
> > It is not requesting for any permissions. Should I perform any action for it to show up?
> 
> Changed it to show popup for existing users on resume of app

I got the popup on new install too on my real device S10+ after notification permission is granted. Clicking on "Enable Setting" took me to this screen and do not know what to do next as I don't see what I should enable 😀

**Comment by adeekshith** [comment]
Remaining Todos in this PR:
- [ ] Remove popup and show "Enable Auto Start" Option in Settings -> Troubleshooting
- [ ] Remove Toast that shows up on callback

Rest everything looks 👌

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -128,4 +134,11 @@
     <string name="app_rating_feedback_dialog_msg">Let us know how we can improve.</string>
     <string name="app_rating_feedback_dialog_mail_button_title">Email Us</string>
     <string name="app_rating_feedback_dialog_telegram_button_title">Telegram</string>
+
+    <!--Autostart Permission Dialog-->
+    <string name="auto_start_permission_dialog_title">AutoStart Permission</string>
+    <string name="auto_start_permission_dialog_message">It\'s recommended to ENABLE AUTOSTART permission in the setting. \n\nOtherwise, Watomatic may ocassionally stop working.
```
Comment (adeekshith) [inline app/src/main/res/values/strings.xml:140]: These can be removed right?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -67,9 +68,14 @@
...
     <string name="key_pref_app_language" translatable="false">pref_app_language</string>
     <string name="pref_show_notification_replied_msg" translatable="false">pref_show_notification_replied_msg</string>
+    <string name="pref_auto_start_permission" translatable="false">pref_auto_start_permission</string>
 
     <!-- app settings -->
     <string name="show_notification_label">Show notification for replied messages</string>
+    <string name="preference_category_general_label">General</string>
+    <string name="preference_category_troubleshooting_label">Troubleshooting</string>
+    <string name="enable_auto_start_label">Enable Auto Start</string>
+    <string name="enable_auto_start_summary">click to launch settings</string>
```
Comment (adeekshith) [inline app/src/main/res/values/strings.xml:78]: Thinking of a better label for this

**Review by adeekshith — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (spuday90) [reply app/src/main/res/values/strings.xml:140]: oh yeah, i missed it

File: app/src/main/java/com/parishod/watomatic/model/utils/AutoStartHelper.java [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+                dialog.dismiss();
+            });
+
+        }
+    }
+
+    private void showAlert(Context context, DialogInterface.OnClickListener onClickListener) {
+        CustomDialog customDialog = new CustomDialog(context);
+        Bundle bundle = new Bundle();
+        bundle.putString(Constants.PERMISSION_DIALOG_TITLE, context.getString(R.string.auto_start_permission_dialog_title));
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/model/utils/AutoStartHelper.java:202]: `R.string.auto_start_permission_dialog_title` and other string are used here.

These are also not needed right? Can we also remove `showAlert()` then?
Can you clean this up if possible @spuday90

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/model/utils/AutoStartHelper.java:202]: hmm the popup does not show up on emulators. Clicking on the preference in "Settings" does not do anything on emulators. But it shows this popup on my real device.

Comment (spuday90) [reply app/src/main/java/com/parishod/watomatic/model/utils/AutoStartHelper.java:202]: We are showing the popup in settings page

**Review by adeekshith — APPROVED** [review]

**Pull request merged**